libsolidity/formal/CHC.cpp

```diff
@@ -1265,7 +1281,8 @@ void CHC::checkAssertTarget(ASTNode const* _scope, CHCVerificationTarget const&
 	{
 		auto it = m_errorIds.find(assertion->id());
 		solAssert(it != m_errorIds.end(), "");
-		unsigned errorId = it->second;
+		solAssert(it->second.size() >= 1, "");
+		unsigned errorId = it->second[0];
```

Comment:
In the test `smtCheckerTests/modifiers/modifier_two_invocations.sol` there were 2 error ids associated with a single `assert` expression. That should be investigated.

Reply:
There's only one path that reaches that assertion, so yea, we should check that.

Reply:
Any conclusion here?

Reply:
Since there are two applications of the same modifier in that example, the `assert` statements is visited twice, once when visiting the inner modifier and once when visiting the outer modifier. My intuition says that there could be a problem there, that this situation is not covered correctly. I will try to craft an example.

Reply:
I think it shouldn't be visited twice, because the modifiers are inlined around the function, so:
```
modifier m {
   statement;
   _;
}
function f() m m {
    assert;
}
```

should be the same as
```
function f() {
    statement;
    statement;
    assert;
}
```

Reply:
That's true, but the `assert` is in the modifier, not in the function. So when you inline them, it gets duplicated to the assertions. But I believe there is only one ASTNode since there is only one assert statement in the original source code.

Reply:
Ah right, my bad. Makes sense then